File: test_dir/cpp/src/tuples/exceptions.cpp
#include "tuples\exceptions.h"

void tuple_exceptions_fn() {
    print(PyStr("TUPLE EXCEPTIONS RESULTS:"));
    PyTup<int, double, PyStr> a = PyTup(2, 2.2, PyStr("b"));
    try {
        a.index(1);
    } catch (const PyppValueError &pypp_e) {
        std::string e = pypp_e.what();
        print(PyStr("value error: ") + to_pystr(e));
    }
}

File: test_dir/python/src/tuples/exceptions.py
def tuple_exceptions_fn():
    print("TUPLE EXCEPTIONS RESULTS:")
    a: tuple[int, float, str] = (2, 2.2, "b")
    try:
        a.index(1)
    except ValueError as e:
        print("value error: " + str(e))
    # NOTE: pypp_tg(a, 3) results in a compile error in the C++ rather than a runtime
    # error.
    # try:
    #     pypp_tg(a, 3)
    # except IndexError as e:
    #     print("index error: " + str(e))
